### PR TITLE
Host - UI: Bug fixes

### PR DESCRIPTION
**Pull request by deetz99**
*Issue:*

- bcgov/entity/issues/25093
- bcgov/entity/issues/25101
- bcgov/entity/issues/25099
- bcgov/entity/issues/25098
- bcgov/entity/issues/25089
- bcgov/entity/issues/25097

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the BC Registry and Digital Services BSD 3-Clause License

### CODE REVIEW

File: strr-base-web/app/components/form/common/Contact.vue [hunk 1]
```diff
@@ -118,6 +118,7 @@ defineProps<{
           :aria-label="t('label.faxNumberOpt')"
           name="faxNumber"
           :placeholder="t('label.faxNumberOpt')"
+          :mask="phone?.countryCode === '1' ? '(###) ###-####' : '##############'"
```
Comment (deetz99) [inline strr-base-web/app/components/form/common/Contact.vue:121]: add input mask to fax number that matches the same mask as the phone input

File: strr-host-pm-web/app/components/form/AddOwners/input/Person.vue [hunk 2]
```diff
@@ -141,7 +141,7 @@ watch(isCompParty, (val) => {
           id="host-owner-taxNumber"
           v-model="owner.taxNumber"
           name="taxNumber"
-          :placeholder="$t('label.craTaxNumberOpt')"
+          :placeholder="$t('label.craTaxNumber')"
```
Comment (deetz99) [inline strr-host-pm-web/app/components/form/AddOwners/input/Person.vue:144]: remove optional from label

File: strr-host-pm-web/app/components/form/AddOwners/input/Person.vue [hunk 3]
```diff
@@ -197,6 +197,7 @@ watch(isCompParty, (val) => {
           :aria-label="$t('label.faxNumberOpt')"
           name="faxNumber"
           :placeholder="$t('label.faxNumberOpt')"
+          :mask="owner.phone.countryCode === '1' ? '(###) ###-####' : '##############'"
```
Comment (deetz99) [inline strr-host-pm-web/app/components/form/AddOwners/input/Person.vue:200]: add input mask to fax number that matches phone number input mask

File: strr-host-pm-web/app/components/form/Review/Index.vue
Comment (deetz99) [inline strr-host-pm-web/app/components/form/Review/Index.vue:1]: only display pr declaration in review confirmation list if pr required and not exempt

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-examiner-dev--pr-425-0zg9axvp.web.app

File: strr-host-pm-web/app/pages/application.vue
Comment (deetz99) [inline strr-host-pm-web/app/pages/application.vue:1]: reset fee room count each change

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-platform-dev--pr-425-gw5nbnny.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-strata-hotel-dev--pr-425-d5osuois.web.app

File: strr-host-pm-web/app/stores/document.ts [hunk 4]
```diff
@@ -65,7 +65,7 @@ export const useDocumentStore = defineStore('host/document', () => {
       })
     }
 
-    if (propStore.unitDetails.ownershipType === OwnershipType.RENT) {
+    if (propStore.unitDetails.ownershipType === OwnershipType.RENT && reqs.isPrincipalResidenceRequired && exemptionReason === undefined) {
```
Comment (deetz99) [inline strr-host-pm-web/app/stores/document.ts:68]: only require rental docs if pr required and not exempt

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-hosts-dev--pr-425-4zstjlap.web.app

File: strr-host-pm-web/app/stores/document.ts [hunk 5]
```diff
@@ -255,11 +255,15 @@ export const useDocumentStore = defineStore('host/document', () => {
   }
 
   function validatePrincipalResidenceDocuments (): boolean {
-    // either 2 unique docs from this list are required
-    const uniqueColumnADocs = [
+    // bc id cards only count as 1 document
```
Comment (deetz99) [inline strr-host-pm-web/app/stores/document.ts:258]: only count bc identification docs as max 1 value

File: strr-host-pm-web/app/stores/hostOwner.ts [hunk 6]
```diff
@@ -32,7 +32,9 @@ export const useHostOwnerStore = defineStore('host/owner', () => {
       dateOfBirth: type === OwnerType.INDIVIDUAL && role === OwnerRole.HOST
         ? getRequiredNonEmptyString(t('validation.dateOfBirth'))
         : optionalOrEmptyString,
-      taxNumber: getOptionalSin(t('validation.sin'))
+      taxNumber: type === OwnerType.INDIVIDUAL && role === OwnerRole.HOST
```
Comment (deetz99) [inline strr-host-pm-web/app/stores/hostOwner.ts:35]: make sin required

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-examiner-dev--pr-425-0zg9axvp.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-platform-dev--pr-425-gw5nbnny.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-strata-hotel-dev--pr-425-d5osuois.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-hosts-dev--pr-425-4zstjlap.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-examiner-dev--pr-425-0zg9axvp.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-platform-dev--pr-425-gw5nbnny.web.app

File: strr-host-pm-web/app/components/form/Review/Index.vue [hunk 7]
```diff
@@ -65,6 +66,14 @@ const getCompPartyName = computed(() => {
   }
   return `[${t('label.completingParty')}]`
 })
+
+const agreedToRentalActListItems = computed(() => {
```
Comment (deetz99) [inline strr-host-pm-web/app/components/form/Review/Index.vue:70]: only display pr declaration if pr required and not exempt

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-strata-hotel-dev--pr-425-d5osuois.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-hosts-dev--pr-425-4zstjlap.web.app

File: strr-host-pm-web/app/pages/application.vue
Comment (deetz99) [inline strr-host-pm-web/app/pages/application.vue:1]: reset fees between changes

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**